.github/workflows/release.yml

```diff
@@ -31,24 +31,55 @@ jobs:
         run: |
           echo "::set-output name=version::$(git describe --tags --always --dirty | cut -c2-)"
 
-  builder:
-    needs: [args]
+  # Based on https://github.com/google/osv-scanner/blob/124ad0e2311dc8f25cec2aa298e467a7fffef3b1/.github/workflows/goreleaser.yml#L11-L53
+  goreleaser:
+    outputs:
+      hashes: ${{ steps.hash.outputs.hashes }}
+    permissions:
+      contents: write  # for goreleaser/goreleaser-action to create a GitHub release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: 1.18
+      - name: Run GoReleaser
+        id: run-goreleaser
+        uses: goreleaser/goreleaser-action@8f67e590f2d095516493f017008adc464e63adb1 # v4.1.0
```

Comment:
would it be possible to improve the slsa provenance generation of Go releaser first?
Goreleaser currently does not generate provenance, and the Action downloads the binary without verification https://github.com/goreleaser/goreleaser-action/blob/master/src/goreleaser.ts#L24-L41

It's a been of a chicken-and-egg problem as well. Take a look at what we do in the slsa-verifier Action https://github.com/slsa-framework/slsa-verifier/blob/main/actions/installer/src/index.ts#L165-L184

Reply:
Thank you for your review.
I know GoReleaser has an issue about SLSA. https://github.com/goreleaser/goreleaser/issues/2737
But the progress of this issue is unclear so I think we can use slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml until GoReleaser will support generating SLSA Provenance.

Reply:
I think this change doesn't reduce the guarantee of slsa-verifier.
Do you have a concern goreleaser-action is tampered?

Reply:
I got it. Thanks.

Reply:
Thanks. Are you still interested in helping out set up the workflow with a strategy matrix?

Reply:
Yeah, I agree a strategy matrix would be the best for slsa-verifier to reduce attack vectors. We need to be pretty strict with this repo since it's a security focused project.

Reply:
re: Goreleaser provenance.
We know it takes time and it's time-consuming to write and maintain a reusable-workflow-based builder. To help with that, we're working on a new feature "(Bring / Build Your Own Builder) BYOB" https://github.com/slsa-framework/slsa-github-generator/milestone/11

This framework will allow existing GitHub Actions to be "wrapped" into a reusable workflow with minimum effort or maintenance. We plan to finish it in Q1, and we're working with several teams to use the framework to build npm and Java builders.

We'd be very interested in showing the team you how to make use of it for GoReleaser, and to migrate all the users of the generic generator + GoReleaser to a simpler-to-use and more official GoReleaser builder.